src/pages/AccountSettings/tabs/OrgUploadToken/TokenlessSection/TokenlessSection.tsx

```diff
@@ -1,25 +1,52 @@
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
 
+import { useUploadTokenRequired } from 'services/uploadTokenRequired'
 import A from 'ui/A'
 import { Card } from 'ui/Card'
 import { RadioTileGroup } from 'ui/RadioTileGroup'
 
-import TokenlessModal from './TokenRequiredModal'
+import TokenRequiredModal from './TokenRequiredModal'
+import { useSetUploadTokenRequired } from './useSetUploadTokenRequired'
 
 const AUTHENTICATION_OPTIONS = {
   NotRequired: 'not-required',
   Required: 'required',
 } as const
 
-function TokenlessSection() {
-  const [showModal, setShowModal] = useState(false)
-  const [tokenRequired, setTokenRequired] = useState(false) // TODO: get from API
+interface UseParams {
+  provider: string
+  owner: string
+}
+
+const TokenlessSection: React.FC = () => {
+  const { provider, owner } = useParams<UseParams>()
+  const {
+    data: uploadTokenRequiredData,
+    isLoading: isUploadTokenRequiredLoading,
+  } = useUploadTokenRequired({ provider, owner })
+  const { mutate, isLoading: isSetUploadTokenRequiredLoading } =
+    useSetUploadTokenRequired()
+
+  const [showModal, setShowModal] = useState<boolean>(false)
+  const [tokenRequired, setTokenRequiredState] = useState<boolean>(true)
+
+  useEffect(() => {
+    if (
+      !isUploadTokenRequiredLoading &&
+      uploadTokenRequiredData &&
+      uploadTokenRequiredData?.uploadTokenRequired !== null
```

Comment:
nit: if we're checking for uploadTokenRequiredData I think we don't need the ? on uploadTokenRequiredData?.uploadTokenRequired

Reply:
TS is weird, this actually tests if it's set to boolean, if i just do uploadTokenRequiredData?.uploadTokenRequired !== null and set it to `uploadTokenRequiredData?.uploadTokenRequired` it complains about a nullable value: 
<img width="1512" alt="Screenshot 2024-10-10 at 5 29 32 AM" src="https://github.com/user-attachments/assets/360ad4ad-4c99-4bdd-8df7-1670410b59d6">
<img width="1512" alt="Screenshot 2024-10-10 at 5 29 29 AM" src="https://github.com/user-attachments/assets/31330dfa-6284-4e9a-a862-de8dc6ff162d">